joomla/joomla-cms
# [4.0] Separate core.js into multiple files

Do NO SQUASH merge this commit please.

There are 2 reasons for this PR. First get core.js more module and split into multiple files for better maintainability. The second is a copyright issue reported by @brianteeman for the webcomponents part of the core.js

### Summary of Changes
Make it possible to split concat javascript files into a main file based on the directory name.
Add change history in git commits to preserve copyright and change information.


### Testing Instructions
you need to run npm install,
test Joomla core functionality like
* javascript translations
* check error  messages creation in forms / ajax forms
* check webcomponents loaded correctly  


### Expected result
works


### Actual result
works
